internal/resources/policy/kind/image/recipe/allowed_name_tag_schema.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package recipe
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	policyrecipeimagemodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/policy/recipe/image"
+)
+
+var AllowedNameTag = &schema.Schema{
+	Type:        schema.TypeList,
+	Description: "The input schema for image policy allowed-name-tag recipe version v1",
+	Optional:    true,
+	ForceNew:    true,
+	MaxItems:    1,
+	Elem: &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			AuditKey: {
+				Type:        schema.TypeBool,
+				Description: "Audit (dry-run). Violations will be logged but not denied.",
+				Optional:    true,
+				Default:     false,
+			},
+			RulesKey: {
+				Type:        schema.TypeList,
+				Description: "It specifies a list of rules that defines allowed image patterns.",
+				Required:    true,
+				MinItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						ImageNameKey: {
+							Type:        schema.TypeString,
```

Comment:
Optional without explicit default value?

Reply:
Added default value which is empty string at all applicable places.